### PR TITLE
Added support for directives

### PR DESCRIPTION
**Pull request by alanaasmaa**
Example:

```
'use stricts'
const GraphQLError = use('Adonis/Addons/GraphQLError')

module.exports = {
  async isAuthenticated (next, src, args, { auth }) {
    try {
      await auth.check()
      return next()
    } catch (error) {
      throw new GraphQLError('User has to be authenticated')
    }
  }
}
```

There can be issues I started to learn GraphQL yesterday.

### CODE REVIEW

**Comment by alanaasmaa** [comment]
I don't like much of the idea to create another `Directives` folder but it's better than nothing.

**Comment by RomainLanz** [comment]
Hey @alanaasmaa! 👋 

Thanks for this contribution! One thing, it seems that you got an issue with your history.
Could you try to fix it by "rebasing" and hard pushing?

**Comment by alanaasmaa** [comment]
Hey @RomainLanz,

I assumed you are gonna ask this 😅 
I'm not very comfort by rebasing but I have to learn that skill at some point.

I appreciate very much what you have done to make AdonisJS a great framework.

File: commands/MakeDirectives.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+ * @license MIT
+ * @copyright Slynova - Romain Lanz <romain.lanz@slynova.ch>
+ */
+
+const _ = require('lodash')
+const Config = use('Config')
+const { join, sep } = require('path')
+const { Command } = require('@adonisjs/ace')
+
+class MakeSchema extends Command {
```
Comment (yariksav) [inline commands/MakeDirectives.js]: can you rename to MakeDirectives?

Comment (alanaasmaa) [reply commands/MakeDirectives.js]: Forgot to mention that i renamed it.

File: commands/MakeDirectives.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    const templateContent = await this.readFile(templatePath, 'utf-8')
+
+    await this.generateFile(filePath, templateContent, { name })
+
+    const createdFile = filePath.replace(process.cwd(), '').replace(sep, '')
+    console.log(`${this.icon('success')} ${this.chalk.green('create')}  ${createdFile}`)
+  }
+}
+
+module.exports = MakeSchema
```
Comment (yariksav) [inline commands/MakeDirectives.js]: this also has to be "MakeDirectives"

Comment (alanaasmaa) [reply commands/MakeDirectives.js]: @yariksav Done.

**Comment by alanaasmaa** [comment]
I think this pull request also need one change.
Currently, I assume it's not backward compatible. Those who update via npm/yarn will get errors when they do not create Directives folder. Should we make the folder optional?

**Comment by yariksav** [comment]
Really need this PR with directive's logic. And so long time... maybe in 2020 it appears?

**Comment by RomainLanz** [comment]
@yariksav You know that this PR only add a command to create a directive with a boilerplate? 🙄

**Comment by alanaasmaa** [comment]
This PR is also not backwards compatible I havent had time to continue my AdonisJs project so i havent finished this.

But its start.

**Comment by yariksav** [comment]
My apologizes, @RomainLanz, I've opened #8 issue, but you've closed it by this PR.
This module allows register only schema and resolves
https://github.com/RomainLanz/adonis-graphql/blob/29e9561ed64e338c03879a94b7067904fbd217dd/src/Server/index.js#L19
but function makeExecutableSchema  receives other arguments, such as directiveResolvers, ...

Therefore, for realization directives in adonis I should **make stub in hooks**
```js
hooks.after.providersBooted(() => {
  const { attachDirectiveResolvers } = require('graphql-tools')
  const GraphQLServer = use('Adonis/Addons/GraphQLServer')

  const directiveResolvers = {
   // my directive resolves
  }
  attachDirectiveResolvers(
    GraphQLServer.$schema,
    directiveResolvers
  );
})
```

**Comment by RomainLanz** [comment]
Cool, I'll take a look this weekend and release a dev version so you can try it out and tell me if it's working well 👍

**Comment by yariksav** [comment]
Deal! Thank you, @RomainLanz!

**Comment by PieterVenter7** [comment]
Hey is there any progress on this or a way I can help with the testing?

**Pull request closed without merging**